### PR TITLE
Stock items job

### PR DESCRIPTION
**Pull request by BenMorganIO**
In the previous PR, some bright guy (me) decided to merge master into the stock items job PR. That resulted in it being closed by Github after a revert and a hard reset. This PR takes off where #5888 left off.

This PR includes quite a bit of refactoring in the specs while searching for reasons behind some Fs.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://benmorganio-stock-items-job.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://benmorganio-stock-items-job.spree-spree-9ed724b77adec04c49c2.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: core/spec/models/spree/customer_return_spec.rb [hunk 1]
```diff
@@ -113,36 +108,31 @@
...
       it "should mark all inventory units are returned" do
         create(:customer_return_without_return_items, return_items: [return_item], stock_location_id: inventory_unit.shipment.stock_location_id)
         expect(inventory_unit.reload.state).to eq 'returned'
       end
 
       it "should update the stock item counts in the stock location" do
-        expect do
-          create(:customer_return_without_return_items, return_items: [return_item], stock_location_id: inventory_unit.shipment.stock_location_id)
-        end.to change { inventory_unit.find_stock_item.count_on_hand }.by(1)
+        create(:customer_return_without_return_items, return_items: [return_item], stock_location_id: inventory_unit.shipment.stock_location_id)
```
Comment (BenMorganIO) [inline core/spec/models/spree/customer_return_spec.rb]: That's better than the 146...

File: core/spec/spec_helper.rb [hunk 2]
```diff
@@ -4,10 +4,10 @@
   SimpleCov.start do
     add_group 'Controllers', 'app/controllers'
     add_group 'Helpers', 'app/helpers'
+    add_group 'Jobs', 'app/jobs'
     add_group 'Mailers', 'app/mailers'
     add_group 'Models', 'app/models'
     add_group 'Views', 'app/views'
-    add_group 'Jobs', 'app/jobs'
     add_group 'Libraries', 'lib'
```
Comment (JDutil) [inline core/spec/spec_helper.rb]: Libraries can be alphabetized too.

File: core/app/models/spree/stock_location.rb [hunk 3]
```diff
@@ -12,8 +12,8 @@ class StockLocation < Spree::Base
     scope :active, -> { where(active: true) }
     scope :order_default, -> { order(default: :desc, name: :asc) }
 
-    after_create :create_stock_items, :if => "self.propagate_all_variants?"
     after_save :ensure_one_default
+    after_commit :create_stock_items, on: :create, if: "self.try :propagate_all_variants?"
```
Comment (JDutil) [inline core/app/models/spree/stock_location.rb]: I'm not sure I like making this after_commit the stock location should properly be created so you shouldn't need the try.  I feel like this let's things possibly fail silently.

Comment (BenMorganIO) [inline core/app/models/spree/stock_location.rb]: I couldn't figure out how to get the tests passing without it...

I'm thinking of moving the seed data inside of the migrations out into the seeds.

Comment (JDutil) [inline core/app/models/spree/stock_location.rb]: Could have the seeds destroy the migrated data and seed fresh records if seeds are run.

**Comment by JDutil** [comment]
Could you please resubmit against master?  Didn't notice this before merging 4.2 branch.

**Pull request closed without merging**